--- run-agent.sh	2018-07-23 23:52:50.000000000 +0200
+++ /dev/fd/63	2018-08-18 01:18:39.033307555 +0200
@@ -74,9 +74,7 @@
    sleep 1
 done

-trap "${AGENT_DIST}/bin/agent.sh stop; exit 0;" SIGINT SIGTERM SIGHUP
+trap '${AGENT_DIST}/bin/agent.sh stop; while ps -p $(cat $(ls -1 ${LOG_DIR}/*.pid)) &>/dev/null; do sleep 1; done; kill %%' SIGINT SIGTERM SIGHUP

-touch /root/anchor
-
-tail -qF ${LOG_DIR}/teamcity-agent.log /root/anchor &
+tail -qF ${LOG_DIR}/teamcity-agent.log &
 wait
